src/daft-plan/src/logical_ops/filter.rs

```diff
@@ -18,7 +18,8 @@ pub struct Filter {
 
 impl Filter {
     pub(crate) fn try_new(input: Arc<LogicalPlan>, predicate: ExprRef) -> Result<Self> {
-        let (predicate, field) = resolve_expr(predicate, &input.schema()).context(CreationSnafu)?;
+        let (predicate, field) =
```

Comment:
what happens if you filter using a wildcard? 

Within a filter context, I'd intuitively expect that a `col('*')` would be equivalent of an `any` on all of the columns 

```py
df = daft.from_pydict(
    {
        "a": [1, 2, 3],
        "b": ['a', 'b', 'c']
    }
)

df.where(col('*') > 1) 
# equivalent to
df.where(any([col(column) for column in df.columns)]) 
```

Reply:
Right now it just errors since `where` doesn't accept multiple expressions at once.

I don't think making it `any` is necessarily the best choice--to me it also seems intuitive for `col('*') > 1` to be equivalent to an `all`. I think we should keep `any` or `all` explicit (once we do add those).